src/use-mutation.ts

```diff
@@ -23,6 +23,26 @@ export interface UseMutationOptions<
    */
   keys?: _MutationKeys<TParams, TResult>
 
+  /**
+   * Hook to execute a callback when the mutation is triggered
+   */
+  onMutate?: (...args: TParams) => void
```

Comment:
It should be able to return a context (or a promise of it) that is later passed to other hooks

```suggestion
  onMutate?: (...args: TParams) => Promise<TContext> | TContext
```